vars/default.yml

```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: MIT
+---
+# Put internal variables here with values for unrecognized distributions.
+
+# Example:
+__template_packages: []
+__template_services: []
```

Comment:
not sure if you will need these - if so, you might need to have different variables for both platform and provider e.g. you might need files like `vars/certmonger_RedHat_7.yml`, `vars/self_signed_Fedora.yml`, etc. - but I wouldn't do that until it becomes necessary - for now, you might only need provider specific files.

Reply:
Is that really required to split deps in files by provider/distro? 
That's how I've done so far: 0569e988dba550ec6af56df9c1718422069040ef
For global providers I see the point but for task specific providers I can't really see any advantage.

Reply:
Basically "__certificate_packages" will vary depending on OS version.

In case of certmonger it's a single service and in case of openssl no service will be required so for now I think we can consider that service doesn't need to be a list.

Shouldn't the defaults.yml and main.yml together contain all variables and the other files (by OS) just overwrite them for doc purpose? I do like when a role is constructed this way because it's really easy to find what was used where.

Reply:
New attempt: c85e509de0b98157e5968d72ca56b5b673cad9c8

Reply:
> 

I think the way it works is that `defaults/main.yml` and `vars/main.yml` are always applied.  Then, you can do something like this https://github.com/linux-system-roles/cockpit/blob/master/tasks/main.yml#L1-L8 to add or replace platform specific variables.

Reply:
> In case of certmonger it's a single service and in case of openssl no service will be required so for now I think we can consider that service doesn't need to be a list.

Ok, but I think all of the other roles implement support for a list of services, even if there is only 1 service, so this will be inconsistent with other roles.

Reply:
Alright. I can do same way to keep consistency.

It's just kinda of weird to add code that it's probably not going to be executed anytime soon and will just make the code harder to understand :)

Reply:
I think having 1 service is ok - just add a comment to the effect of "yes, I know every other role has a list of services, but this one does not"

Reply:
Added. I've done a rebase because there were too many fixups already.